Common/TableProducer/occupancyTableProducer.cxx

```diff
@@ -0,0 +1,1053 @@
+    if (collisions.size() == 0) {
+      return;
+    }
+    //
+    auto bc = collisions.iteratorAt(0).bc_as<myBCTable>();
+    int run = bc.runNumber();
+    auto time = bc.timestamp();
+
+    auto runDuration = ccdb->getRunDuration(run, true);
+    int64_t tsSOR = runDuration.first;
+    auto ctpx = ccdb->getForTimeStamp<std::vector<Long64_t>>("CTP/Calib/OrbitReset", tsSOR);
+    int64_t tsOrbitReset = (*ctpx)[0];
+    uint32_t nOrbitsPerTF = run < 534133 ? 128 : 32;
+    int64_t orbitSOR = (tsSOR * 1000 - tsOrbitReset) / o2::constants::lhc::LHCOrbitMUS;
+    orbitSOR = orbitSOR / nOrbitsPerTF * nOrbitsPerTF;
+    int64_t bcSOR = orbitSOR * nBCsPerOrbit + customOrbitOffset * nBCsPerOrbit;
+    int32_t nBCsPerTF = nOrbitsPerTF * nBCsPerOrbit;
+    // nBCsPerITSROF = (run >= 543437 && run <= 545367) ? 594 : 198;
+
+    int64_t TFidThis = (bc.globalBC() - bcSOR) / nBCsPerTF;
+    int bcInTF = (bc.globalBC() - bcSOR) % nBCsPerTF;
+    //
+
+    std::vector<float> Occ_Prim_Unfm_80;
+    std::vector<float> Occ_FV0A_Unfm_80;
+    std::vector<float> Occ_FV0C_Unfm_80;
+    std::vector<float> Occ_FT0A_Unfm_80;
+    std::vector<float> Occ_FT0C_Unfm_80;
+    std::vector<float> Occ_FDDA_Unfm_80;
+    std::vector<float> Occ_FDDC_Unfm_80;
+
+    std::vector<float> Occ_NTrack_PVC_Unfm_80;
+    std::vector<float> Occ_NTrack_ITS_Unfm_80;
+    std::vector<float> Occ_NTrack_TPC_Unfm_80;
+    std::vector<float> Occ_NTrack_TRD_Unfm_80;
+    std::vector<float> Occ_NTrack_TOF_Unfm_80;
+    std::vector<float> Occ_NTrackSize_Unfm_80;
+    std::vector<float> Occ_NTrackTPC_A_Unfm_80;
+    std::vector<float> Occ_NTrackTPC_C_Unfm_80;
+    std::vector<float> Occ_NTrackITS_TPC_Unfm_80;
+    std::vector<float> Occ_NTrackITS_TPC_A_Unfm_80;
+    std::vector<float> Occ_NTrackITS_TPC_C_Unfm_80;
+
+    std::vector<float> OccRobust_T0V0Prim_Unfm_80;
+    std::vector<float> OccRobust_FDDT0V0Prim_Unfm_80;
+    std::vector<float> OccRobust_NtrackDet_Unfm_80;
+
+    int64_t oldTFid = -1;
+
+    int64_t oldCollisionIndex = -100;
+    int CollisionId_negError = 0;
+    int CollisionId_posError = 0;
+    int CollisionId_NoError = 0;
+    int nAmbgTracks = 0;
+    bool hasCollision = false;
+    bool isAmbgTrack = false;
+    bool LastTrackHadCollision = false;
+    bool doCollisionUpdate = false;
+    bool doAmbgUpdate = false;
+
+    double Rbegin = 90., Rend = 245.;
+    double Zbegin; // collision.posZ() + track.tgl()*Rbegin;
+    double Zend;   // collision.posZ() + track.tgl()*Rend;
+    double vdrift = 2.64;
+
+    double dTbegin; // ((250.- TMath::Abs(Zbegin))/vdrift)/0.025;//bin
+    double dTend;   // ((250.- TMath::Abs(Zend))/vdrift)/0.025;  //bin
+
+    double bcBegin; // tGlobalBC + dTbegin;
+    double bcEnd;   // tGlobalBC + dTend  ;
+
+    int BinBCbegin;
+    int BinBCend;
+
+    for (const auto& trackQA : tracksQA) {
+      // auto track = trackQA.track_as<myTracks>;  // dereferncing not working // only option is either use iterator way or Table slicing
```

Comment:
I do not see any reason why dereferencing would not work in this case, what is the error you are getting?

Reply:
It worked, thank you pointing out the mistake here